packages/runtime/src/next/client.ts

```diff
@@ -140,33 +140,71 @@ export type MakeswiftPageSnapshot = {
 
 const makeswiftComponentDocumentSchema = z.object({
   id: z.string(),
-  type: z.string(),
-  key: z.string(),
   name: z.string().nullable(),
   locale: z.string().nullable(),
   data: Schema.element,
+  siteId: z.string(),
 })
 
 export type MakeswiftComponentDocument = z.infer<typeof makeswiftComponentDocumentSchema>
 
-export type MakeswiftComponentSnapshot = {
-  document: MakeswiftComponentDocument
-  cacheData: CacheData
-}
+const makeswiftComponentDocumentFallbackSchema = z.object({
+  id: z.string(),
+  locale: z.string().nullable(),
+  data: z.null(),
+  apiKeyPrefix: z.string(),
+})
+
+export type MakeswiftComponentDocumentFallback = z.infer<
+  typeof makeswiftComponentDocumentFallbackSchema
+>
+
+export type MakeswiftComponentSnapshot =
+  | {
+      document: MakeswiftComponentDocument
+      cacheData: CacheData
+    }
+  | {
+      document: MakeswiftComponentDocumentFallback
+      cacheData: CacheData | null
```

Comment:
Instead of `cacheData | null` we should use empty cache data. Maybe we can have a factory for that, e.g., `CacheData.empty()`